pennylane/labs/trotter_error/product_formulas/bch.py

```diff
@@ -491,3 +514,21 @@ def _drop_zeros(
             del terms[commutator]
 
     return term_dicts
+
+
+def _drop_unimportant(terms, importance, tolerance):
+    delete = []
+    for commutator, coeff in terms.items():
+        if any(isinstance(frag, ProductFormula) for frag in commutator):
+            continue
+
+        if np.abs(coeff) * _commutator_importance(commutator, importance) < tolerance:
+            delete.append(commutator)
+
+    for commutator in delete:
+        del terms[commutator]
+
+
+def _commutator_importance(commutator, importance):
+    commutator_importance = [importance[x] for x in commutator]
+    return 2 ** (len(commutator) - 1) * reduce(lambda x, y: x * y, commutator_importance)
```

Comment:
Please explain in the docstring where this logic is taken from, e.g., a paper reference or a brief explanation.